### PR TITLE
Replace all global TypeError instances in JsBuiltins

### PR DESCRIPTION
**Pull request by meg-gupta**

### CODE REVIEW

File: lib/Runtime/Library/JsBuiltIn/JsBuiltIn.js [hunk 1]
```diff
@@ -120,7 +122,7 @@
     platform.registerFunction(FunctionsEnum.ArrayEntries, function () {
         "use strict";
         if (this === null || this === undefined) {
-            throw new TypeError("Array.prototype.entries: 'this' is null or undefined");
+            __chakraLibrary.raiseThis_NullOrUndefined("Array.prototype.entries");
         }
```
Comment (meg-gupta) [inline lib/Runtime/Library/JsBuiltIn/JsBuiltIn.js]: I could not use platform.raiseThis_NullOrUndefined the same was as Intl uses, Intl maybe doing something special to hold reference to the platform object defined in the parent initialization closure code. Adding error functions on the always available chakraLibrary object.

Comment (meg-gupta) [reply lib/Runtime/Library/JsBuiltIn/JsBuiltIn.js]: @curtisman does this change look good to unblock you for now ?

**Comment by meg-gupta** [comment]
@dotnet-bot test OSX static_osx_osx_debug

File: lib/Runtime/Library/JsBuiltIn/JsBuiltIn.js (hunk 1 above)
Comment (curtisman) [reply lib/Runtime/Library/JsBuiltIn/JsBuiltIn.js]: yep

---
In reply to: [179288292](https://github.com/Microsoft/ChakraCore/pull/4934#discussion_r179288292) [](ancestors = 179288292)

**Comment by akroshg** [comment]
Are there any test case to reach the code path?

**Comment by meg-gupta** [comment]
@akroshg es6 tests cover all the error cases.

**Comment by akroshg** [comment]
Sounds good.

**Pull request merged**